src/main/java/io/vertx/gsoc2018/qotd/QuoteOfTheDayVerticle.java

```diff
@@ -32,9 +33,13 @@ public void start(Future<Void> startFuture) {
     Future<Void> dbReady = Future.future();
     Future<Void> httpServerReady = Future.future();
 
-    // complete startFuture only when db and http server started
-    compose(startFuture, dbReady, httpServerReady);
-    compose(startFuture, httpServerReady, dbReady);
+    CompositeFuture.all(dbReady, httpServerReady).setHandler(result -> {
```

Comment:
This is concurrent composition. We don't want everything to be started  concurrently, we want these steps, in order:

1. setup the schema
2. import base data
3. start the http server.

In your example, the http server could start reply requests without the DB being ready

Reply:
Ah, got it.

Reply:
Just changed, so now the steps are done sequentially